test/evergreen/compatibility_test_for_mongodb_releases.sh

```diff
@@ -1,131 +1,134 @@
+
+        cd "$1/test/format"
+
+        args=""
+        args+="cache=80 "                       # Medium cache so there's eviction
+        args+="checkpoints=1 "                  # Force periodic writes
+        args+="compression=snappy "             # We only built with snappy, force the choice
+        args+="data_source=table "
+        args+="in_memory=0 "                    # Interested in the on-disk format
+        args+="leak_memory=1 "                  # Faster runs
+        args+="logging_compression=snappy "     # We only built with snappy, force the choice
+        args+="rebalance=0 "                    # Faster runs
+        args+="rows=1000000 "
+        args+="salvage=0 "                      # Faster runs
+        args+="timer=4 "
+        args+="verify=0 "                       # Faster runs
+
+        for am in $2; do
+            dir="RUNDIR.$am"
+            echo "./t running $am access method..."
+            ./t -1q -h $dir "file_type=$am" $args
+        done
 }
 
-run()
+EXT="extensions=["
+EXT+="ext/compressors/snappy/.libs/libwiredtiger_snappy.so,"
+EXT+="ext/collators/reverse/.libs/libwiredtiger_reverse_collator.so, "
+EXT+="ext/encryptors/rotn/.libs/libwiredtiger_rotn.so, "
+EXT+="]"
+
+#############################################################
+# verify_backward:
+#       arg1: release #1
+#       arg2: release #2
+#       arg3: access methods list
+#############################################################
+verify_backward()
 {
-	# Build test files from each release.
-	(build_rel 3.4) || return 1
-	(build_rel 3.6) || return 1
-	(build_rel 4.0) || return 1
-	(build_rel 4.2) || return 1
-	(build_rel develop) || return 1
-
-	# Verify forward/backward compatibility.
-	(verify 3.4 3.6) || return 1
-	(verify 3.6 4.0) || return 1
-	(verify 4.0 4.2) || return 1
-	(verify 4.2 develop) || return 1
-
-	return 0
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+        echo "Release \"$1\" verifying \"$2\""
+        echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+        
+        cd "$1"
+        for am in $3; do
+            dir="$2/test/format/RUNDIR.$am"
+            echo "$1/wt verifying $2 access method $am..."
+
+            WIREDTIGER_CONFIG="$EXT" ./wt -h "../$dir" verify table:wt
+        done
 }
 
 # Create a directory in which to do the work.
 top="test-compatibility-run"
-rm -rf $top && mkdir $top && cd $top || {
-	echo "$0: unable to create $top working directory"
-	exit 1
-}
-
-run
-exit $?
+rm -rf "$top" && mkdir "$top"
+cd "$top"
+
+# Build a release with a known name, then use it to figure out tagged release names.
+(build_release "develop")
+
+# Figure out the names of the releases we'll compare.
+cd develop
+rel34="$(get_tagged_release "3.4")"
+rel36="$(get_tagged_release "3.6")"
+rel40="$(get_tagged_release "4.0")"
+#rel42="$(get_tagged_release "4.2")"    XXX cannot handle 4.4 formats yet.
+rel42="mongodb-4.2"
```

Comment:
Why not setting `rel42` the same as other release versions?

Reply:
Because no tagged 4.2 release version can handle the current 4.4 formats, you have to use the `mongodb-4.2` branch instead of the most recent tagged 4.2 release. Once we tag the current `mongodb-4.2` branch with a release number, we can fix this.

Reply:
Alternatively, we could switch to using the `mongodb-XX` branches instead of (or in addition too), the release branches.

Reply:
I thought we are going to use a 4.4 binary to verify 4.2 data files? There are tagged 4.2 release versions (e.g. 4.2.5-rc1) but no tagged 4.4 version (the 1st 4.4.0-rc0 has not been cut). I guess you mean once we tag the current `mongodb-4.4` branch we can add the backward compatibility test for 4.4/4.2.

Reply:
> I thought we are going to use a 4.4 binary to verify 4.2 data files?

Yes, and for now I'm using the `develop` branch as the stand-in for the 4.4 release.

We could use the `mongodb-4.4` branch instead, of course.

I think the argument for using release-tagged branches is we're testing exactly what goes out in the field.

The argument for using `develop` and `mongodb-X.X` branches is we're testing sooner rather than later, and we can catch problems faster.

Preferences?

Or, we could do both.

Reply:
"Catching problem faster" sounds more useful than "finding a released version broke compatibility requirement".

Reply:
@lukech, commit 1163d42 switches from testing tagged releases to testing named branch releases.

This version is not likely to pass Evergreen testing until we merge the `develop` branch into the `mongodb-4.4` branch, but then I think it will work.  If you can't do that update now, let me know and I can adjust the test to avoid those problems.

Also, even though we've restricted the `4.4/develop` branch testing to row-store, running format does occasionally fail, so it's not out of the question for this test to go red in Evergreen. Although it shouldn't be for any reason to do with compatibility, we'll still need to triage the failure.

Reply:
Thanks for making the change @keithbostic. It may take some time before `mongodb-4.4` becomes stable - thats the stage we are able to sync `develop` with `mongodb-4.4` branch. So I'd prefer to disabling develop/mongodb-4.4 compatibility test for the time being, and re-enable it later on with a follow-on ticket.

The occasional format failure is not a big concern to me. The triage baron should be able to attribute failures to existing bug tickets (or create new ones as needed).

Reply:
@lukech, commit a32d67113 removes `mongodb-4.4` branch from the script, testing the `develop` branch instead.